internal/jimmhttp/httpproxy_handler.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Canonical.
+
+package jimmhttp
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/juju/names/v4"
+	"gopkg.in/errgo.v1"
+
+	"github.com/canonical/jimm/v3/internal/jimm"
+	"github.com/canonical/jimm/v3/internal/middleware"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+	"github.com/canonical/jimm/v3/internal/rpc"
+)
+
+// HTTPProxyHandler is an handler that provides proxying capabilities.
+// It uses the uuid in the path to proxy requests to model's controller.
+type HTTPProxyHandler struct {
+	Router *chi.Mux
+	jimm   *jimm.JIMM
+}
+
+const (
+	// all endpoints managed by this handler
+	ProxyEndpoints = "/*"
+)
+
+// NewHTTPProxyHandler creates a proxy http handler.
+func NewHTTPProxyHandler(jimm *jimm.JIMM) *HTTPProxyHandler {
+	return &HTTPProxyHandler{Router: chi.NewRouter(), jimm: jimm}
+}
+
+// Routes returns the grouped routers routes with group specific middlewares.
+func (hph *HTTPProxyHandler) Routes() chi.Router {
+	hph.SetupMiddleware()
+	hph.Router.HandleFunc(ProxyEndpoints, hph.ProxyHTTP)
+	return hph.Router
+}
+
+// SetupMiddleware applies authn and authz middlewares.
+func (hph *HTTPProxyHandler) SetupMiddleware() {
```

Comment:
Should this be exported?

Reply:
It is to comply with HTTPHandler interface.
https://github.com/canonical/jimm/blob/85c03a97fbe54830e306706cca4174bff1f379bc/internal/jimmhttp/handler.go#L8-L12

Reply:
I think we should edit the interface. None of the structs that implement it, call it outside of the Routes() method. Which makes sense, If a struct has some required middleware, that is internal to the implementation of the handler.

Additionally, one could wrap the handler in more middleware.

Reply:
addressing this in a followup